### PR TITLE
Misc: queries optimisation for refresh service

### PR DESCRIPTION
**Pull request by annvelents**

### CODE REVIEW

File: app/jobs/clock/finalize_invoices_job.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class FinalizeInvoicesJob < ApplicationJob
     unique :until_executed, on_conflict: :log
 
     def perform
-      Invoice.ready_to_be_finalized.find_each do |invoice|
+      Invoice.ready_to_be_finalized.includes(:credit_notes).find_each do |invoice|
```
Comment (nudded) [inline app/jobs/clock/finalize_invoices_job.rb:12]: How does this include help? 🤔

Comment (annvelents) [reply app/jobs/clock/finalize_invoices_job.rb:12]: inside `FinlizeJob` we pass this invoice to `Invoices::RefreshDraftAndFinalizeService`, where there we're passing it further to `RefreshDraftService`, where finally we're going through 
```
        cn_subscription_ids = invoice.credit_notes.map do |cn|
          {credit_note_id: cn.id, subscription_id: cn.fees.pick(:subscription_id)}
        end
```
and actually in addition to the credit notes we can also preload fees... 🤔

Comment (nudded) [reply app/jobs/clock/finalize_invoices_job.rb:12]: but the inclusion here does nothing right? we load the invoice again in that job?

File: db/migrate/20241217120924_add_indexes_to_invoices.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AddIndexesToInvoices < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    add_index :invoices, :ready_to_be_refreshed, where: "ready_to_be_refreshed", algorithm: :concurrently
```
Comment (nudded) [inline db/migrate/20241217120924_add_indexes_to_invoices.rb:7]: I'd be explicit `where: "ready_to_be_refreshed = true"`

File: app/models/invoice.rb [hunk 3]
```diff
@@ -105,7 +105,7 @@ class Invoice < ApplicationRecord
   scope :visible, -> { where(status: VISIBLE_STATUS.keys) }
   scope :invisible, -> { where(status: INVISIBLE_STATUS.keys) }
   scope :with_generated_number, -> { where(status: %w[finalized voided]) }
-  scope :ready_to_be_refreshed, -> { where(ready_to_be_refreshed: true) }
+  scope :ready_to_be_refreshed, -> { draft.where(ready_to_be_refreshed: true) }
```
Comment (nudded) [inline app/models/invoice.rb:108]: Definitely need a spec for this scope I think

**Review by nudded — APPROVED** [review]

**Pull request merged**